bug: CI is failing

CI is failing on main https://github.com/AI4quantum/maestro/actions/runs/16995647521/job/48505487311

This seems to have been caused by some upstream dependency since this test passed on main 2 days ago but now it's failing.

```
service/mcp-fetch-proxy patched
service/mcp-fetch-proxy patched
  + Exception Group Traceback (most recent call last):
  |   File "/home/runner/work/maestro/maestro/tests/utils/mcpclient.py", line 23, in <module>
  |     asyncio.run(main())
  |   File "/home/runner/.local/share/uv/python/cpython-3.12.8-linux-x86_64-gnu/lib/python3.12/asyncio/runners.py", line 194, in run
  |     return runner.run(main)
  |            ^^^^^^^^^^^^^^^^
  |   File "/home/runner/.local/share/uv/python/cpython-3.12.8-linux-x86_64-gnu/lib/python3.12/asyncio/runners.py", line 118, in run
  |     return self._loop.run_until_complete(task)
  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |   File "/home/runner/.local/share/uv/python/cpython-3.12.8-linux-x86_64-gnu/lib/python3.12/asyncio/base_events.py", line 686, in run_until_complete
  |     return future.result()
  |            ^^^^^^^^^^^^^^^
  |   File "/home/runner/work/maestro/maestro/tests/utils/mcpclient.py", line 8, in main
  |     async with streamablehttp_client("http://127.0.0.1:30051/mcp") as (
  |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |   File "/home/runner/.local/share/uv/python/cpython-3.12.8-linux-x86_64-gnu/lib/python3.12/contextlib.py", line 231, in __aexit__
  |     await self.gen.athrow(value)
  |   File "/home/runner/work/maestro/maestro/.venv/lib/python3.12/site-packages/mcp/client/streamable_http.py", line 474, in streamablehttp_client
  |     async with anyio.create_task_group() as tg:
  |                ^^^^^^^^^^^^^^^^^^^^^^^^^
  |   File "/home/runner/work/maestro/maestro/.venv/lib/python3.12/site-packages/anyio/_backends/_asyncio.py", line 772, in __aexit__
  |     raise BaseExceptionGroup(
  | ExceptionGroup: unhandled errors in a TaskGroup (1 sub-exception)
  +-+---------------- 1 ----------------
    | Traceback (most recent call last):
    |   File "/home/runner/work/maestro/maestro/.venv/lib/python3.12/site-packages/mcp/client/streamable_http.py", line 405, in handle_request_async
    |     await self._handle_post_request(ctx)
    |   File "/home/runner/work/maestro/maestro/.venv/lib/python3.12/site-packages/mcp/client/streamable_http.py", line 277, in _handle_post_request
    |     response.raise_for_status()
    |   File "/home/runner/work/maestro/maestro/.venv/lib/python3.12/site-packages/httpx/_models.py", line 829, in raise_for_status
    |     raise HTTPStatusError(message, request=request, response=self)
    | httpx.HTTPStatusError: Server error '502 Bad Gateway' for url 'http://127.0.0.1:30051/mcp'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/502
    +------------------------------------
Error: Process completed with exit code 1.
```

--- a/.github/workflows/maestro_create_tool-test.yaml
+++ b/.github/workflows/maestro_create_tool-test.yaml
@@ -24,8 +24,8 @@ jobs:
         kind create cluster --config tests/integration/deploys/kind-config.yaml 
     - name: Install ToolHive
       run: |
-        helm upgrade -i toolhive-operator-crds oci://ghcr.io/stacklok/toolhive/toolhive-operator-crds
-        helm upgrade -i toolhive-operator oci://ghcr.io/stacklok/toolhive/toolhive-operator -n toolhive-system --create-namespace
+        helm upgrade -i toolhive-operator-crds oci://ghcr.io/stacklok/toolhive/toolhive-operator-crds --version 0.0.13
+        helm upgrade -i toolhive-operator oci://ghcr.io/stacklok/toolhive/toolhive-operator -n toolhive-system --create-namespace --version 0.2.1
     - name: Set up Python
       uses: actions/setup-python@v5
       with:

--- a/.github/workflows/maestro_mcp_tool_test.yaml
+++ b/.github/workflows/maestro_mcp_tool_test.yaml
@@ -24,8 +24,8 @@ jobs:
         kind create cluster --config tests/integration/deploys/kind-config.yaml 
     - name: Install ToolHive
       run: |
-        helm upgrade -i toolhive-operator-crds oci://ghcr.io/stacklok/toolhive/toolhive-operator-crds
-        helm upgrade -i toolhive-operator oci://ghcr.io/stacklok/toolhive/toolhive-operator -n toolhive-system --create-namespace
+        helm upgrade -i toolhive-operator-crds oci://ghcr.io/stacklok/toolhive/toolhive-operator-crds --version 0.0.13
+        helm upgrade -i toolhive-operator oci://ghcr.io/stacklok/toolhive/toolhive-operator -n toolhive-system --create-namespace --version 0.2.1
     - name: Set up Python
       uses: actions/setup-python@v5
       with:

--- a/.github/workflows/maestro_mcp_tool_test.yaml
+++ b/.github/workflows/maestro_mcp_tool_test.yaml
@@ -1,4 +1,4 @@
-name: Maestro MCP tool Test
+name: Maestro MCP Tool Test
 on:
   push:
     branches: [ "main" ]

--- a/.github/workflows/maestro_mcp_tool_test.yaml
+++ b/.github/workflows/maestro_mcp_tool_test.yaml
@@ -1,4 +1,4 @@
-name: Maestro Create tool Test
+name: Maestro MCP tool Test
 on:
   push:
     branches: [ "main" ]

